### PR TITLE
Revert "ClickHouse: Normalize one batch at a time"

### PR DESCRIPTION
**Pull request by heavycrystal**
Reverts PeerDB-io/peerdb#2219

seems to degrade perf

### CODE REVIEW

**Review by iskakaushik — APPROVED** [review]

File: flow/connectors/external_metadata/store.go [hunk 1]
```diff
@@ -176,9 +176,7 @@ func (p *PostgresMetadata) UpdateNormalizeBatchID(ctx context.Context, jobName s
 		`UPDATE `+lastSyncStateTableName+
 			` SET normalize_batch_id=$2 WHERE job_name=$1`, jobName, batchID)
 	if err != nil {
-		p.logger.Error("failed to update normalize batch id",
-			slog.Any("error", err),
-			slog.Int64("batchID", batchID))
+		p.logger.Error("failed to update normalize batch id", slog.Any("error", err))
```
Comment (serprex) [inline flow/connectors/external_metadata/store.go:179]: can keep this logging

File: flow/connectors/clickhouse/normalize.go [hunk 2]
```diff
@@ -486,11 +435,60 @@ func (c *ClickHouseConnector) syncTablesInThisBatch(
...
 		if err := c.execWithLogging(ctx, q); err != nil {
-			return fmt.Errorf("error while inserting into normalized table: %w", err)
+			return nil, fmt.Errorf("error while inserting into normalized table: %w", err)
 		}
 	}
 
-	return nil
+	err = c.UpdateNormalizeBatchID(ctx, req.FlowJobName, req.SyncBatchID)
+	if err != nil {
+		c.logger.Error("[clickhouse] error while updating normalize batch id", "error", err)
```
Comment (serprex) [inline flow/connectors/clickhouse/normalize.go:444]: should log SyncBatchID

**Review by serprex — DISMISSED** [review]

File: flow/connectors/clickhouse/normalize.go [hunk 3]
```diff
@@ -486,11 +435,60 @@ func (c *ClickHouseConnector) syncTablesInThisBatch(
...
+
+		if !tableName.Valid {
+			return nil, errors.New("table name is not valid")
+		}
+
+		tableNames = append(tableNames, tableName.String)
+	}
+
+	if rows.Err() != nil {
+		return nil, fmt.Errorf("failed to read rows: %w", err)
```
Comment (serprex) [inline flow/connectors/clickhouse/normalize.go:488]:
```suggestion
	if err := rows.Err(); err != nil {
		return nil, fmt.Errorf("failed to read rows: %w", err)
```

File: flow/connectors/clickhouse/normalize.go [hunk 4]
```diff
@@ -486,11 +435,60 @@ func (c *ClickHouseConnector) syncTablesInThisBatch(
...
+	rows, err := c.query(ctx, q)
+	if err != nil {
+		return nil, fmt.Errorf("error while querying raw table for distinct table names in batch: %w", err)
+	}
+	defer rows.Close()
+	var tableNames []string
+	for rows.Next() {
+		var tableName sql.NullString
+		err = rows.Scan(&tableName)
+		if err != nil {
```
Comment (serprex) [inline flow/connectors/clickhouse/normalize.go:476]:
```suggestion
		if err := rows.Scan(&tableName); err != nil {
```

**Comment by heavycrystal** [comment]
will address review feedback as part of rebasing #2248 
changes during reverts might be confusing

**Pull request merged**